content/yaml-quick-start/building-a-vr-oculus-app.md

```diff
@@ -0,0 +1,345 @@
+
+<h2 id="unity-build-script">Add a build script to Unity</h2>
+
+A Unity build script is required to build the project in headless mode.
+
+Open Unity and add a new C# script in the project explorer in Assets/Editor called **Build**.
+
+Paste the following script into the new file:
+
+```cs
+using System.Linq;
+using System;
+using UnityEditor;
+using UnityEngine;
+
+public static class BuildScript
+{
+
+    [MenuItem("Build/Build Android")]
+    public static void BuildAndroid()
+    {
+
+        PlayerSettings.Android.useCustomKeystore = true;
+        EditorUserBuildSettings.buildAppBundle = false;
+
+        // Set keystore name
+        string keystoreName = Environment.GetEnvironmentVariable("CM_KEYSTORE_PATH");
+        if (!String.IsNullOrEmpty(keystoreName))
+        {
+            Debug.Log($"Setting path to keystore: {keystoreName}");
+            PlayerSettings.Android.keystoreName = keystoreName;
+        }
+        else
+        {
+            Debug.Log("Keystore name not provided");
+        }
+
+        // Set keystore password
+        string keystorePass = Environment.GetEnvironmentVariable("CM_KEYSTORE_PASSWORD");
+        if (!String.IsNullOrEmpty(keystorePass))
+        {
+            Debug.Log("Setting keystore password");
+            PlayerSettings.Android.keystorePass = keystorePass;
+        }
+        else
+        {
+            Debug.Log("Keystore password not provided");
+        }
+
+        // Set keystore alias name
+        string keyaliasName = Environment.GetEnvironmentVariable("CM_KEY_ALIAS");
+        if (!String.IsNullOrEmpty(keyaliasName))
+        {
+            Debug.Log("Setting keystore alias");
+            PlayerSettings.Android.keyaliasName = keyaliasName;
+        }
+        else
+        {
+            Debug.Log("Keystore alias not provided");
+        }
+
+        // Set keystore password
+        string keyaliasPass = Environment.GetEnvironmentVariable("CM_KEY_PASSWORD");
+        if (!String.IsNullOrEmpty(keyaliasPass))
+        {
+            Debug.Log("Setting keystore alias password");
+            PlayerSettings.Android.keyaliasPass = keyaliasPass;
+        }
+        else
+        {
+            Debug.Log("Keystore alias password not provided");
+        }
+        BuildPlayerOptions buildPlayerOptions = new BuildPlayerOptions();
+        buildPlayerOptions.locationPathName = "android/android.apk";
+        buildPlayerOptions.target = BuildTarget.Android;
+        buildPlayerOptions.options = BuildOptions.None;
+        buildPlayerOptions.scenes = GetScenes();
+
+        Debug.Log("Building Android");
+        BuildPipeline.BuildPlayer(buildPlayerOptions);
+        Debug.Log("Built Android");
+    }
+
+    private static string[] GetScenes()
+    {
+        return (from scene in EditorBuildSettings.scenes where scene.enabled select scene.path).ToArray();
+    }
+
+}
+```
+
+**Warning:** this script does not auto-increment the bundle version of your Android app so you will need to:
```

Comment:
Why not use the workflow build number (BUILD_NUMBER)  or project build number (PROJECT_BUILD_NUMBER) from Codemagic to increment `bundleVersionCode`? 

Does Oculus expect a different `bundleVersion` for each submission, or can that remain the same but with a different `bundleVersionCode`?

I would have thought you could just do this so no additional C# scripts have to be added:

```        // Set bundle version
        var versionIsSet = int.TryParse(Environment.GetEnvironmentVariable("BUILD_NUMBER"), out int version);
        if (versionIsSet)
        {
            Debug.Log($"Bundle version code set to {version}");
            PlayerSettings.Android.bundleVersionCode = version;
        }
        else
        {
            Debug.Log("Bundle version not provided");
        }
```

Reply:
Hi @nevercode-kevin ! So yes it needs to be different each time, just like when you publish on the Play Store for example... but you're right, we can just increase the last code digit I guess. The important thing is that the full code (eg. 1.0.2 changes each time).
(And contrary to Play Store, the Oculus CLI tool has no command to get the version of the latest build sadly :/)

Thanks for the suggestion, I'll update the docs! :)

Reply:
Just to clarify, what does Oculus expect? For example, is it ok to have version 1.0.0 and then build 1, 2, 3 of the same version? Or can it have the same build number but a different version e.g. version 1.0.1 (build number 1), version 1.0.2 (build number 1)?

Reply:
Hey! So, according to these docs: https://developer.oculus.com/resources/publish-mobile-manifest/#unity
It looks like the important part is indeed to increase the **Bundle Version Code** and not necessarily the entire **Bundle Version**.
So I'm guessing something like `1.0.2b0 -> 1.0.2.b1` would be fine :)
(but please tell me if I'm not reading their docs right and you feel it won't work!)

Reply:
Yes, that would be fine.